build/docs.js

```diff
@@ -184,30 +189,44 @@ function traverseRAMLResourceTree (resources, callback, absURL) {
  * @param  {Object} $ plugin loader
  * @return {Stream}
  */
-module.exports = (gulp, $) => {
+module.exports = (gulp) => {
     gulp.task('java-clone', (callback) => {
         getRepo('git@github.com:WatchBeam/beam-client-java.git', callback);
     });
     gulp.task('java-mvn-gen', ['java-clone'], (callback) => {
-        childProcess.exec(`cd ${config.src.tmp}/beam-client-java && mvn clean javadoc:javadoc`, callback);
+        childProcess.exec(
+            `cd ${config.src.tmp}/beam-client-java && mvn clean javadoc:javadoc`,
+            callback
+        );
     });
     gulp.task('java-doc', ['java-mvn-gen'], () => {
         return gulp.src(path.join(config.src.tmp, 'beam-client-java/target/site/apidocs'))
         .pipe(gulp.dest(config.dist.javadoc));
     });
 
     gulp.task('backend-clone', (callback) => {
+        if (config.backendRamlPath) {
+            callback();
+            return;
+        }
         getRepo('git@github.com:WatchBeam/backend.git', callback);
     });
     gulp.task('backend-doc', ['backend-clone'], () => {
-        return ramlParser().loadApi(
-            path.join(config.src.tmp, 'backend/doc/raml/index.raml'),
-            { rejectOnErrors: true }
-        )
+        let docPath;
+        if (config.backendRamlPath) {
+            docPath = path.join(config.backendRamlPath, 'index.raml');
+        } else {
+            docPath = path.join(config.src.tmp, 'backend/doc/raml/index.raml');
+        }
+        return ramlParser().loadApi(docPath, {
+            rejectOnErrors: true,
+        })
         .catch(error => {
             if (error.parserErrors) {
                 const stack = error.parserErrors
-                .map((err, idx) => `${idx + 1}: ${err.path}@${err.line}:${err.column} ${err.message}`)
+                .map((err, idx) =>
+                    `${idx + 1}: ${err.path}@${err.line}:${err.column} ${err.message}`
+                )
                 .join('\n');
                 error.message += `\n${stack}`;
```

Comment:
I feel like those should be added on a case-by-case basis, javadoc build is pretty much independent from the developer docs, if you want a preview of those it's much faster to generate inside the java project itself.